docs/service/processor.rst

```diff
@@ -50,13 +44,42 @@ All helper scripts run in the shell in the container::
 
     $ make shell
 
+Some of the scripts require downloading production data from
+`crash-stats.mozilla.com <https://crash-stats.mozilla.com>`_, and it is
+useful to add an API token before entering the shell.
+
+
+.. _`API token`:
+
+Adding an API Token
+-------------------
+The data download scripts require an API token with the "View Raw Dumps"
+permission to acquire raw crash data. This crash data will not include
+personally identifiable information (PII). Without PII, the processor will not
+add useful symbols or signatures. This public-only crash data is useful for
+Socorro development, not for debugging crash reports. Additional permissions
+are required to include PII in the crash data.
+
+You can generate API tokens at `<https://crash-stats.mozilla.com/api/tokens/>`_.
+
+.. Note::
+
+   Make sure you treat any data you pull from production in accordance with our
+   data policies that you agreed to when granted access to it.
+
+Add the API token value to your ``my.env`` file::
+
+    SOCORRO_API_TOKEN=apitokenhere
+
+The API token is used by the data download scripts (run inside
+``$ make shell``), but not directly by the processor.
 
 process_crashes.sh
 ------------------
 
 You can use the ``scripts/process_crashes.sh`` script which will fetch crash
 data, sync it with the S3 bucket, and publish the crash ids to Pub/Sub
-for processing.
+for processing. This command requires a valid `API token`_.
```

Comment:
I tried to do these commands so that they don't *require* a valid API token. You can still do things without one. The API token only helps if you have additional privileges--they let you pull down PII parts of the crash data. The old docs are muddled on this point.

I think I'd rephrase this as "If you have an account that has access to memory dumps, be sure to use an API token so you can download that data." Or something like that.

Reply:
I thought ``View Raw Dumps`` was required, but it appears raw crash data is downloaded with ``SOCORRO_API_TOKEN=`` (blank token). I'm going to try to download some data so I'm clear on what the permissions do.

Reply:
OK it looks like you either need no API key, or an API key with both permissions. I'll update the docs to remove the middle tier of permissions.

**Blank token**
* Prints "No api token provided. Skipping dumps and personally identifiable information."
* Prints "Working on [uuid]..."
* ``[crashdata]/v2/raw_crash/[prefix]/[date]/[uuid]`` gets sanitized JSON data
* ``[crashdata]/v1/dump_names/[uuid]`` gets an empty list ``[]``

**Token with "View Raw Dumps"**
* Prints "Using api token: abcdxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
* Prints "Working on [uuid]..."
* ``[crashdata]/v2/raw_crash/[prefix]/[date]/[uuid]`` gets sanitized JSON data
* ``[crashdata]/v1/dump_names/[uuid]`` gets an empty list ``[]``

**Token with "View Personal Identifiable Information"**
* Prints "Using api token: abcdxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
* Prints "Working on [uuid]..."
* ``[crashdata]/v2/raw_crash/[prefix]/[date]/[uuid]`` gets complete JSON data
* Exception raised ``Exception: Something unexpected happened. status_code 403, content b"Binary response requires the 'View Raw Dumps' permission\n"``

**Token with both permissions**
* Prints "Using api token: abcdxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
* Prints "Working on [uuid]..."
* ``[crashdata]/v2/raw_crash/[prefix]/[date]/[uuid]`` gets complete JSON data
* ``[crashdata]/v1/dump_names/[uuid]`` gets a 1-item list ``["[the name]"]``
* ``[crashdata]/v1/dump/[dump uuid]`` gets a binary dump

Reply:
This is goofy. The two permissions seem inextricably entwined. Seems like we should combine them or dump one or something.

Having said that, this is the only place where the awkwardness shows up that I can think of. Fiddling with permissions and such is non-trivial, so as a stop-gap, maybe we add a note to the top of the tokens page and push any other changes off until a rainy day.

https://crash-stats.mozilla.com/api/tokens/

Maybe something like:

"If you're creating an API token for downloading crash data, you need 'View Raw Dumps' and 'View Personal Identifiable Information'."

Reply:
People don't read the docs first?!? 😲

I think I'll tackle non-doc stuff in a new PR.